src/uwtools/drivers/forecast.py

```diff
@@ -138,82 +132,60 @@ def requirements(self) -> None:
         ???
         """
 
-    def resources(self, platform: dict) -> Mapping:
+    def resources(self) -> Mapping:
         """
         Parses the config and returns a formatted dictionary for the batch script.
         """
-        # Add required fields to platform.
-        # Currently supporting only slurm scheduler.
+
         return {
-            "account": platform["account"],
-            "nodes": 1,
-            "queue": "batch",
-            "scheduler": self.job_scheduler,
-            "tasks_per_node": 1,
-            "walltime": "00:01:00",
+            "account": self._experiment_config["user"]["account"],
+            "scheduler": self._experiment_config["platform"]["scheduler"],
+            **self._config["jobinfo"],
         }
 
-    def run(self) -> None:
+    def run(self, cycle: datetime) -> None:
         """
         Runs FV3 either as a subprocess or by submitting a batch script.
         """
-        # Read in the config file.
-        forecast_config = self._config["forecast"]
-        platform_config = self._config["platform"]
-
         # Prepare directories.
-        run_directory = forecast_config["RUN_DIRECTORY"]
+        run_directory = self._config["run_dir"]
         self.create_directory_structure(run_directory, "delete")
 
-        static_files = forecast_config["STATIC"]
-        self.stage_files(run_directory, static_files, link_files=True)
-        cycledep_files = forecast_config["CYCLEDEP"]
-        self.stage_files(run_directory, cycledep_files, link_files=True)
-
-        # Create the job script.
-        platform_resources = self.resources(platform_config)
-        batch_script = self.batch_script(platform_resources)
-        args = "--export=None"
-        run_command = self.run_cmd(
-            args,
-            run_cmd=platform_config["MPICMD"],
-            exec_name=forecast_config["EXEC_NAME"],
-        )
-        batch_script.append(run_command)
+        self._config["cycle-dependent"].update(self._define_boundary_files())
 
-        if self._dry_run:
-            # Apply switch to allow user to view the run command of config.
-            # This will not run the job.
-            logging.info("Batch Script:")
-            logging.info(batch_script)
-            return
+        for file_category in ["static", "cycle-dependent"]:
```

Comment:
Which one should we adopt? I agree standardization is good.

Reply:
I had seen hyphens in examples before and stuck with it, but apparently YAML can go either way. I did see a suggestion to stick with the conventions of the native language, which would be underscores for Python and it seems reasonable enough for me

Reply:
I prefer hyphens, for exactly the opposite reason that @WeirAE cites: Using hyphens serves as a reminder that YAML is _not_ Python and that there's no connection between the YAML keys in our DSL and values in other languages. (This may also help remind people that `None` is not to be used as a YAML value -- it will deserialize to the Python _string_ `"None"`, not the Python `None` object. For better or for worse, the Pythonic values `True` and `False` are also YAML boolean values, so that's less of a concern.)

But none of this matters until/unless we decide that we want a DSL where the keyspace is defined by the language. I won't restate my arguments for that here, but unless we go that route, we *must* use underscores for keys, because keys may need to be meaningful in other languages -- e.g. bash. Again, I'm not in favor of this, but it's the direction we're pointed in at the moment.

Reply:
Unfortunately, we're in the strangling phase using some bash and some Python drivers. We may need to translate keys in certain sections to environment variables for use in bash scripts.

Reply:
Do those values have to be keys, though? Can't they be values instead, e.g.
```
environment-variables:
  - name: SOME_ENV_VAR
    value: 88
  - name: ANOTHER_ENV_VAR
    value: foo
```
As long as we export `SOME_ENV_VAR=88 ANOTHER_ENV_VAR=foo` to the legacy drivers, they shouldn't care how we represent those in UW YAML.

Reply:
I don't hate that approach. I can move toward hyphens. We can't be consistent with that whole rule anyway, given that the need for filenames to be keys is creeping in for other cases...

Reply:
...I'm going to open a GH Issue on this so we don't forget.

Reply:
> the need for filenames to be keys is creeping in for other cases...
I'll have to be convinced of such a "need". I don't see it.